File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/UsersDownloadedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.User;

import java.util.HashMap;

/**
 * Created by Evan on 6/9/15.
 */
public class UsersDownloadedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        UsersDownloadedEventCallback.class.getClassLoader();
    public static final Creator<UsersDownloadedEventCallback> CREATOR =
        new Creator<UsersDownloadedEventCallback>() {
            @Override
            public UsersDownloadedEventCallback createFromParcel(Parcel in) {
                return new UsersDownloadedEventCallback(
                    (User) in.readValue(CL), (HashMap<String, User>) in.readValue(CL));
            }

            @Override
            public UsersDownloadedEventCallback[] newArray(int size) {
                return new UsersDownloadedEventCallback[size];
            }
        };
    public final HashMap<String, User> users;

    public UsersDownloadedEventCallback(User loggedInUser, HashMap<String, User> users) {
        super(loggedInUser);
        this.users = users;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(users);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/UsersDownloadedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable
import com.shareyourproxy.api.domain.model.User
import java.util.*

/**
 * User contacts downloaded.
 */
internal final class UsersDownloadedEventCallback(loggedInUser: User, val users: HashMap<String, User>) : UserEventCallback(loggedInUser) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(users)
    }

    companion object {
        private val CL = UsersDownloadedEventCallback::class.java.classLoader
        val CREATOR: Parcelable.Creator<UsersDownloadedEventCallback> = object : Parcelable.Creator<UsersDownloadedEventCallback> {
            @Suppress("UNCHECKED_CAST")
            override fun createFromParcel(parcel: Parcel): UsersDownloadedEventCallback {
                return UsersDownloadedEventCallback(parcel.readValue(CL) as User, parcel.readValue(CL) as HashMap<String, User>)
            }

            override fun newArray(size: Int): Array<UsersDownloadedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
